backend/de.metas.business/src/main/java-legacy/org/compiere/process/M_PriceList_Create.java

```diff
@@ -22,11 +28,31 @@
  * @author metas-dev <dev@metasfresh.com>
  *
  */
-public class M_PriceList_Create extends JavaProcess
+public class M_PriceList_Create extends JavaProcess implements IProcessPrecondition
 {
 	// Services
 	private final transient ISessionBL sessionBL = Services.get(ISessionBL.class);
 	private final transient IAttributeDAO attributeDAO = Services.get(IAttributeDAO.class);
+	private  final transient IPriceListDAO priceListDAO = Services.get(IPriceListDAO.class);
+
+	@Override
+	public ProcessPreconditionsResolution checkPreconditionsApplicable(final @NonNull IProcessPreconditionsContext context)
+	{
+		final PriceListVersionId priceListVersionId = PriceListVersionId.ofRepoIdOrNull(context.getSingleSelectedRecordId());
+		if (priceListVersionId == null)
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection().toInternal();
+		}
```

Comment:
```suggestion
		if (context.isMoreThanOneSelected())
		{
			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection().toInternal();
		}
		final PriceListVersionId priceListVersionId = PriceListVersionId.ofRepoIdOrNull(context.getSingleSelectedRecordId());
```

because `context.getSingleSelectedRecordId()`, based on its javadoc, will throw an error in case of multiple selection